### PR TITLE
feat: integrate with keycloak

### LINKED ISSUE #37: Mattermost: Integration with keycloak
As an app developer, I want to be logged into Mattermost after I log into UDS.

### PR DESCRIPTION
**Pull request by ericwyles**
## Description

Updates to enable sso by default using SSO Secret Templating using the sso spec on the UDS package CR. 

## Related Issue

Fixes #37 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-package-mattermost/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

**Comment by ericwyles** [comment]
It works now without having to manually configure anything, just go to chat.uds.dev, click login with gitlab, redirects you to keycloak. Register yourself a user account and verify email and then chat with yourself.
 
image

File: chart/templates/mattermost-sso.yaml [hunk 1]
```diff
@@ -1,16 +1,28 @@
-apiVersion: v1
-kind: Secret
+apiVersion: uds.dev/v1alpha1
+kind: Package
```
Comment (Racer159) [inline chart/templates/mattermost-sso.yaml:2]: This should probably just go into the existing `uds-package.yaml` so we maintain one per package.

File: chart/values.yaml [hunk 2]
```diff
@@ -16,16 +16,15 @@ postgres:
...
 sso:
-  enabled: false
-  client_id: ""
-  client_secret: ""
-  auth_endpoint: ""
-  token_endpoint: ""
-  user_api_endpoint: ""
+  enabled: true
+  auth_endpoint: "https://sso.###ZARF_VAR_DOMAIN###/realms/uds/protocol/openid-connect/auth"
+  token_endpoint: "https://sso.###ZARF_VAR_DOMAIN###/realms/uds/protocol/openid-connect/token"
```
Comment (Racer159) [inline chart/values.yaml:21]: The token endpoint should be able to just use the internal service name (like GitLab does) - userinfo might be able to as well though YMMV - GitLab cannot use the internal url for that.

File: tasks.yaml [hunk 3]
```diff
@@ -39,6 +39,14 @@ tasks:
       - task: dependencies:create
       - task: create:test-bundle
 
+  - name: local-test
+    description: Test the health of a Mattermost deployment on an existing cluster (for iteration/dev speed)
+    actions:
+      - task: create-mm-test-bundle
+      - task: deploy:test-bundle
+      - task: test:health-check
+      - task: test:ingress
```
Comment (Racer159) [inline tasks.yaml:48]: I think this is fine but we would want to settle on a name - local-test is probably ok but noticed that sonarqube's PR currently calls it: test-package-app-only

Comment (Racer159) [reply tasks.yaml:48]: Eventually this task would use `uds dev deploy` IMO once that is more ready and could be `dev-test` or `test-dev` as well.

**Review by Racer159 — CHANGES_REQUESTED** [review]

File: tasks.yaml (hunk 3 above)
Comment (ericwyles) [reply tasks.yaml:48]: Yep I will rename the sonarqube one to local-test.

File: chart/values.yaml (hunk 2 above)
Comment (ericwyles) [reply chart/values.yaml]: It worked for token but userinfo got a 401 trying to call through the internal service name.

File: tasks.yaml (hunk 3 above)
Comment (zachariahmiller) [reply tasks.yaml:48]: `dev-bundle` 👀

Comment (ericwyles) [reply tasks.yaml]: Renamed it to just `dev` for now and took out the test: tasks

**Review by zachariahmiller — APPROVED** [review]
looks good!

**Pull request merged**